paasta_tools/setup_marathon_job.py

```diff
@@ -242,6 +243,11 @@ def log_bounce_action(line, level='debug'):
 
     kill_given_tasks(client=client, task_ids=[task.id for task in tasks_to_kill], scale=True)
 
+    for task in bounce_lib.flatten_tasks(old_app_at_risk_tasks):
+        if task in tasks_to_kill:
+            hostname = task.host
+            reserve_all_resources([hostname])
```

Comment:
If the bounce code hasn't killed these tasks yet, it will simply reserve whatever resources are available on the host at the time (which is still better than what it currently does). The next time a task is killed from the host, it will try to reserve all the free resources again. There are other places we can add calls to `reserve_all_resources` (i.e. cron job running on the leader that calls `reserve_all_resources` on any host marked as `draining`), but I'd like to hold off on that for now to avoid hammering the REST endpoints too much and to see how this works in practice.